packages/starlight/__tests__/basics/head.test.ts

```diff
@@ -91,9 +91,10 @@ describe('createHead', () => {
 		const defaultFavicon = {
 			tag: 'link',
 			attrs: {
-				rel: 'shortcut icon',
+				rel: 'icon',
 				href: '/favicon.svg',
 				type: 'image/svg+xml',
+				'data-favicon': 'default',
```

Comment:
Thank you for the PR @techfg!

Could you explain why the `data-favicon` attribute was added in this PR? It’s not mentioned in the PR description and I don’t believe it’s a web standard.

Reply:
Ahhh, understood. OK.

Reading the comment in the head processing code, it says:

> The default favicon should be below any extra icons that the user may have set because if several icons are equally appropriate, the last one is used and we want to use the SVG icon when supported.

The current heuristic of relying on `shortcut icon` honestly seems not a great idea (I’m looking at you past self 👀). I wonder if we can just short based on the extension? So instead of needing the extra data attribute, we make sure an SVG favicon is listed last?

Reply:
Agreed on the prior approach not being ideal (e.g., there could be multiple with `shortcut icon` to begin with), was going to mention but wasn't really relevant 😃.

The concern I'd have with detecting SVG directly is that there may not even be an SVG and even if there is, there may be multiple (e.g., config contains custom `head` attributes for different sizes).

I guess the question comes down to what the comment you referenced was really intending to do - I can think of two ways to interpret:

1. It was attempting to ensure that the `favicon` defined in the Starlight configuration property was always last (regardless of its type - e.g., svg, png, ico, jpg, etc.)
2. It was attempting to ensure that any SVG (regardless of whether it was the Starlight configuration property favicon or any other `head` configuration `link`) was last

It appears that [head config](https://github.com/withastro/starlight/commit/c6c1b6727140a76c42c661f406000cc6e9b175de) was added to Starlight prior to [favicon](https://github.com/withastro/starlight/commit/31b8a5aed2bca363c1b05c683b020e596b70bf4a) which provides some context to the comment, but it still is a little ambiguous in terms of what it was intending to do unless I'm just overthinking the comment 😄.

In doing some research on how user agents/browsers determine which browser to use, [Wikipedia](https://en.wikipedia.org/wiki/Favicon#:~:text=Firefox%20and%20Safari%20will%20use,%C3%9716%2C%20otherwise%20the%20ICO.) seemed to be provide some information for anything definitive.  Unfortunately, every browser is a little different assuming the information there is accurate and at the end of the day, user agents/browsers are free to change how they choose a favicon in the future.  

[This article](https://evilmartians.com/chronicles/how-to-favicon-in-2021-six-files-that-fit-most-needs) seems to be the de facto guide on favicons but there is no way at build/runtime for Starlight to know what was added to `head` configuration regarding favicon reliably (or at least without doing a lot of inspection of `rel` & `type` attributes) and it could really be anything with any number of meta tags.

All the above said, do you recall what you were ultimately intending to do with the `favicon` config property and sort weight?  Depending on that, I can think of a few options:

1. Remove the check entirely and just leave at `80`
2. Detect any `link` that is for an `svg` and weight at `70`
3. Detect only the `favicon` config property if it's an `svg` and weight at `70`
4. Detect only the `favicon` config property regardless of type and weight at `70` (this is what the prior code kind of did and what the code in the PR does currently)

One other note is that there is no way to "disable" starlight from generating the favicon link (unless `Head` is overridden) since it seems [merge](https://github.com/withastro/starlight/blob/main/packages/starlight/utils/head.ts#L63) would not detect them as the same (see [stackblitz](https://stackblitz.com/edit/withastro-starlight-kugfxe?file=astro.config.mjs) repro) so there is no way for a user to completely control sort order short of overriding `Head`.  Possibly `favicon` config should support a value of `false` to disable it but that's a separate feature request and may not be necessary.  

In meantime, since the `head` config doesn't provide a way to control sort order of meta tags I think minimizing what is sorted may be best so possibly option 1 should be followed.  If the goal is to control sorting of favicon at some level, I think option 4 may be best since it's basically saying Starlights `favicon` config property will always come last.

Sorry for the long winded comment, this one isn't as straightforward as it may first appear, I think it all comes down to seeing if you can recall why you conditioned the sort weight originally and then deciding from there how to best adjust/handle 😄 

Lemme know your thoughts, thanks!